requirements.txt

```diff
@@ -1,9 +1,7 @@
 pytest >= 4.4
 hypothesis >= 4.18.3
-pytest-xdist
 cffi
 greenlet
-pyparsing
 graphviz
```

Comment:
Cool. Let's definitely do more clean up like this to make pip install much faster

Reply:
I wonder if we can remove graphviz

Reply:
IMO graphviz really should be a developer tool that facilitates debugging. We probably should have different "dependency groups" which expose different set of tools to different audience. For example the "product" group should have core dependencies only, whereas "dev" group might have extra debugging facilitates like graphviz. Python formatters and code coverage tools should probably also go to the "dev" group.

Reply:
hmm maybe we can move "CI dependencies" to .travis.yml?